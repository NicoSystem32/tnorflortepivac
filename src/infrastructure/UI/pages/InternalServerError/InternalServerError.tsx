import { LayoutContent } from '../../transverse'

// Icons
import { Big500SVG } from '../../utils/getIcons'

// Styles
import {
    InternalServerErrorContent,
    InternalServerErrorImg,
    InternalServerErrorInfo,
} from './internalServerError-styles'

const InternalServerError = (): JSX.Element => {
    return (
        <LayoutContent>
            <InternalServerErrorContent className="content-sg">
                <InternalServerErrorImg>
                    <img className="img-server-error" src={Big500SVG} alt="server error" />
                </InternalServerErrorImg>
                <InternalServerErrorInfo>
                    <p className="title-500">Algo salió mal</p>
                    <p className="subtitle-500">
                        En este momento no podemos procesar tu solicitud, por favor intenta de nuevo
                        más tarde, estamos trabajando para solucionarlo.
                    </p>
                </InternalServerErrorInfo>
            </InternalServerErrorContent>
        </LayoutContent>
    )
}
export default InternalServerError
