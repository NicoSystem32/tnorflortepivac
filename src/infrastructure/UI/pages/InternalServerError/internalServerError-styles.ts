import tw, { styled } from 'twin.macro'

export const InternalServerErrorContent = styled.div`
    ${tw`mx-auto mt-33.5 mb-14 min-h-[calc(80vh - 300px)]`}

    @media (min-width: ${({ theme }) => theme.breakpoints.sm}) {
        ${tw`w-[90%]`}
        grid-template-areas: 'info500 img500';
    }

    @media (min-width: ${({ theme }) => theme.breakpoints.md}) {
        ${tw`grid grid-cols-[55% 45%] min-h-[calc(90vh - 300px)]`}
        grid-template-areas: 'info500 img500';
    }

    @media (min-width: ${({ theme }) => theme.breakpoints.lg}) {
        ${tw`mx-auto`}
    }

    @media (min-width: ${({ theme }) => theme.breakpoints.xl}) {
        ${tw`grid w-auto mx-auto grid-cols-[50% 48%]`}
    }
`
export const InternalServerErrorImg = styled.div`
    ${tw`flex flex-col justify-center mb-12.5 h-full bg-no-repeat bg-contain`}
    background-image: url(../../assets/icons/big-stain.png);
    grid-area: img500;

    .img-server-error {
        ${tw`w-auto`}
    }

    @media (min-width: 570px) and (max-width: 1180px) {
        ${tw`mb-auto bg-center`}
        background-size: 90%;

        .img-server-error {
            ${tw`w-full h-80`}
        }
    }

    @media (min-width: 1180px) {
        ${tw`mb-auto w-full bg-right text-center`}
        background-size: 100%;

        .img-server-error {
            ${tw`w-full h-auto`}
        }
    }
`
export const InternalServerErrorInfo = styled.div`
    ${tw`flex flex-col justify-center bg-no-repeat bg-center`}
    background-image: url(../../assets/icons/small-stain.png);
    background-size: 65%;
    grid-area: info500;

    .title-500 {
        ${tw`font-montserrat font-bold text-4xl text-black`}
    }

    .subtitle-500 {
        ${tw`font-helvetica font-normal text-base text-black`}
    }

    @media (min-width: 570px) {
        ${tw`w-[90%]`}
        background-size: 50%;
    }

    @media (min-width: 1180px) {
        ${tw`w-[90%] bg-left`}
        background-size: 40%;

        .title-500 {
            ${tw`text-5xl`}
        }

        .subtitle-500 {
            ${tw`text-base`}
        }
    }
`
