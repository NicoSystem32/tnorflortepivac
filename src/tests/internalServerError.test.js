import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import InternalServerError from '../infrastructure/UI/components/InternalServerError/internalServerError'

describe('Pruebas en <InternalServerError />', () => {
    const wrapper = shallow(<InternalServerError />)

    test('debe de mostrar <InternalServerError /> correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
