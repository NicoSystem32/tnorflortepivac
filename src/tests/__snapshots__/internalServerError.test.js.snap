// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <InternalServerError /> debe de mostrar <InternalServerError /> correctamente 1`] = `
<div
  className="content-sg content-500"
>
  <div
    className="content-img-500"
  >
    <img
      alt=""
      className="img-500"
      src="big-500.png"
      srcSet="big-500.svg"
    />
  </div>
  <div
    className="content-info-500"
  >
    <img
      alt=""
      src=""
    />
    <p
      className="text-gigant-type"
    >
      Algo salió mal
    </p>
    <p
      className="text-secundari-type"
    >
      En este momento no podemos procesar tu solicitud, por favor intenta de nuevo más tarde, estamos trabajando para solucionarlo.
    </p>
  </div>
</div>
`;
